geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java

```diff
@@ -131,9 +171,26 @@ public ResultModel createMapping(
       return ResultModel.createError(ex.getMessage());
     }
 
+    if (pdxClassFile != null) {
+      ManagementAgent agent =
+          ((SystemManagementService) getManagementService()).getManagementAgent();
+      RemoteStreamExporter exporter = agent.getRemoteStreamExporter();
+      remoteInputStreamName = FilenameUtils.getName(tempPdxClassFilePath);
+      remoteInputStream =
+          exporter.export(createSimpleRemoteInputStream(tempPdxClassFilePath));
+    }
+
     CliFunctionResult preconditionCheckResult =
-        executeFunctionAndGetFunctionResult(new CreateMappingPreconditionCheckFunction(), mapping,
+        executeFunctionAndGetFunctionResult(new CreateMappingPreconditionCheckFunction(),
+            new Object[] {mapping, remoteInputStreamName, remoteInputStream},
             targetMembers.iterator().next());
+    if (remoteInputStream != null) {
```

Comment:
is it better to move this code into "finally {}"

Reply:
good idea. I have made that change